WcaOnRails/app/assets/javascripts/application.js

```diff
@@ -46,6 +47,60 @@ wca.cancelPendingAjaxAndAjax = function(id, options) {
   return wca._pendingAjaxById[id];
 };
 
+$.fn.competitionsMap = function(competitions) {
+  var $map = new google.maps.Map(document.getElementById(this.attr('id')), {
+    zoom: 2,
+    center: {lat: 0, lng: 0},
+    scrollwheel: true
+  });
+
+  var markers = [];
+
+  competitions.forEach(function(c) {
+
+    var contentString = "<a href=" + c.url + ">" + c.name + "</a><br />" + c.marker_date + " - " + c.cityName;
+
+    var infowindow = new google.maps.InfoWindow({
+      content: contentString
+    });
+
+    if (c.is_over) {
+      iconImage = 'http://maps.google.com/mapfiles/ms/icons/blue.png';
```

Comment:
Hmmm... is this going to cause a problem when run on an HTTPS site? Browsers don't like mixed content...